src/platform/packages/shared/kbn-elastic-agent-utils/src/agent_sdk_name_and_language.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { isElasticAgentName, isOpenTelemetryAgentName } from './agent_guards';
+
+interface SdkNameAndLanguage {
+  sdkName?: 'apm' | 'edot' | 'otel_other';
+  language?: string;
```

Comment:
Are these two properties really optional?

Reply:
Yes, we have several cases when we return `undefined`: For example custom agent name set, agent name not matching OTel / APM agent names (some could be seen in the [test](https://github.com/elastic/kibana/pull/211822/files/aafbb80b7b3cfe9afc842e8c69b8471f555697f7#diff-3e8c9691b866b603396c811183c3fdb7153cecc8d61b44867c545d09ff79593bR43-R57)): for those cases we can't determine the sdk/language and don't have static dashboards in place so we don't know the values.